src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -58,13 +68,59 @@ def _validation(target, data, evaluators, output_path, tracking_uri, evaluation_
         if not isinstance(evaluation_name, str):
             raise ValueError("evaluation_name must be a string.")
 
+    _validate_columns(data, evaluators, target)
+
+
+def _validate_columns(data, evaluators, target):
     try:
         data_df = pd.read_json(data, lines=True)
     except Exception as e:
-        raise ValueError(f"Failed to load data from {data}. Please validate it is a valid jsonl data. Error: {str(e)}.")
+        raise ValueError(
+            f"Failed to load data from {data}. Please validate it is a valid jsonl data. Error: {str(e)}.")
 
-    for evaluator_name, evaluator in evaluators.items():
-        _validate_input_data_for_evaluator(evaluator, evaluator_name, data_df)
+    if target:
+        # If the target function is given, it may return
+        # several columns and hence we cannot check the availability of columns
+        # without knowing target function semantics.
+        # Instead, here we will validate the columns, taken by target.
+        _validate_input_data_for_evaluator(target, None, data_df, is_target_fn=True)
+    else:
+        for evaluator_name, evaluator in evaluators.items():
+            _validate_input_data_for_evaluator(evaluator, evaluator_name, data_df)
+
+
+def _apply_target_to_data(target: Callable, data: str, pf_client: PFClient) -> str:
+    """
+    Apply the target function to the data set and save the data to temporary file.
+
+    :keyword target: The function to be applied to data.
+    :paramtype target: Callable
+    :keyword data: The path to input jsonl file.
+    :paramtype data: str
+    :keyword pf_client: The promptflow client to be used.
+    :paramtype pf_client: PFClient
+    :return: The path to data file with answers from target function.
+    :rtype: str
+    """
+    with tempfile.TemporaryDirectory() as d:
+        save_function_as_fow(fun=target, target_dir=d, pf=pf_client)
+        run = pf_client.run(
+            flow=d,
+            data=data,
+            name=f'preprocess_{uuid.uuid1()}'
+        )
+        run = pf_client.stream(run)
+        function_output = pd.read_json(pf_client.runs._get_outputs_path(run),
+                                       orient='records', lines=True)
+    function_output.set_index(LINE_NUMBER, inplace=True)
+    function_output.sort_index(inplace=True)
+    data_input = pd.read_json(data, orient='records', lines=True)
+    data_input = pd.concat([data_input, function_output], axis=1, verify_integrity=True)
+    del function_output
+    data_obj = tempfile.TemporaryFile(suffix='.jsonl', mode='w', delete=False)
+    data_obj.close()

```

Comment:
May I missed something. What are these two lines used for?

Reply:
Fixed, we do not actually need the temporary file here.